### DISCUSSION TITLE
ENUM changes

### DISCUSSION THREAD

**Discussion by Roman-Bober** [OpenAPITools/openapi-diff#292] [General]
Hi,
I wonder why enum changes gives "brocken backward compatability"
Shouldn't be if enum is added **at the end of enum list** it should be **backward compatible**?

Thank you,
Roman

**Comment by joschi** [comment]
> Shouldn't be if enum is added **at the end of enum list** it should be **backward compatible**?

I don't think it's backward compatible.

Any client which expects a valid enum value in the response (and validates this) would break when it received a previously unknown enum value.

So in my opinion adding (or removing) any value from an enum is a breaking change.

**Reply by jackbilestech** [reply]
Okay, granted. Then once you have published the enum in an endpoint. Does this not give you a stalemate scenario? In which you can't alter the list at all.

**Reply by joschi** [reply]
> In which you can't alter the list at all.

Sure you can, just not in a backward-compatible way which includes your API and its consumers.
